enhancements/network/routed-ingress-support-for-primary-udn-attached-vms-with-static-ips.md

```diff
@@ -0,0 +1,789 @@
+
+The IPAMClaim CRD status sub-resource will need to be updated, adding
+conditions.
+
+For traceability, we also suggest adding to the `IPAMClaim` status an attribute
+to indicate which pod is holding the claim at any given time. On a VM live
+migration, the OVN-Kubernetes control plane would update the `OwnerPod` after
+the claim has been consumed by a different pod. Same for VM start/stop
+scenarios.
+
+Below you'll see the changes requested:
+```go
+
+// IPAMClaimStatus contains the observed status of the IPAMClaim.
+type IPAMClaimStatus struct {
+    // The list of IP addresses (v4, v6) that were allocated for the pod interface
+    IPs []string `json:"ips"`
++   // The name of the pod holding the IPAMClaim
++   OwnerPod string `json:"ownerPod"`
++   // Conditions contains details for one aspect of the current state of this API Resource
++   Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+```
+
+The `IPAMClaim` status will have (at least) the following conditions:
+- IPAllocated: when the status is "True", it reports the IP address was
+  successfully allocated for the workload. The reason will be
+  `SuccessfulAllocation`. When the status is "False", it reports an IP
+  allocation error, and more details will be provided in the `reason`, and
+  `message` attributes.
+
+These conditions will look like - **successful** allocation:
+```
+type: IPAllocated
+status: "True"
+reason: SuccessfulAllocation
+message: "IP 192.168.1.5 allocated successfully"
+lastTransitionTime: "2025-05-13T11:56:00Z"
+```
+
+These conditions will look like - **failed** allocation:
+```
+type: IPAllocated
+status: "False"
+reason: IPAlreadyExists
+message: "Requested IP 192.168.1.5 is already assigned in the network"
+lastTransitionTime: "2025-05-13T11:56:00Z"
+```
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+The management cluster should have no reason to import an existing VM from
+another virtualization platform into OpenShift Virtualization.
+
+It makes more sense to import VMs into the hosted clusters, in order to provide
+tenants with routed ingress into their hosted cluster workloads.
+
+There should be no hypershift platform-specific considerations with this
+feature.
+
+#### Standalone Clusters
+
+Full support.
+
+#### Single-node Deployments or MicroShift
+
+Full support; there might be some increased resource usage when scaling up
+number of VMs requesting IP and MAC address, since it will mean more processing
+in OVN-Kubernetes, and more CRs to reconcile.
+
+### Implementation Details/Notes/Constraints
+
+TODO
+
+### Risks and Mitigations
+
+TODO
+
+### Drawbacks
+
+The biggest drawback we have to implementing this feature is lack of clear
+asks from customers - we're pretty much guessing what they would want to use.
+
+We do not know (for instance):
+- is this feature only for VMs ?
+- is this feature only about importing VMs ? Should we allow creating new VMs
+  with dedicated MAC / IP / gateway requests ?
```

Comment:
To me it makes sense that whatever is possible for a migrated VM, is surely applicable for the "simpler" case of a new VM (I don't assume the other way around, BTW).
If we are forced to guess the customers needs, then I think we can assume there would be cases where customers prefer avoiding the risk that might be in attempting to copy a VM from one virtualization platform to Openshift (i.e. migration), and create a new VM - with specific MAC/IP/GW - instead.
WDYT?